### ISSUE TITLE
Server image fails to run

### ISSUE THREAD

**Issue by bndw** [couchbase/docker#16]
#### Steps to reproduce
1. `docker run -d -v ~/couchbase:/opt/couchbase/var -p 8091:8091 couchbase/server`

#### Example
```
$ docker run -d -v ~/couchbase:/opt/couchbase/var -p 8091:8091 couchbase/server                                                               
Unable to find image 'couchbase/server:latest' locally
latest: Pulling from couchbase/server
ba2b457ecfb2: Pull complete
26180b0fe8fb: Pull complete
edd7c1974a70: Pull complete
57bca5139a13: Pull complete
0cf7eebde823: Pull complete
14a01c3766b5: Pull complete
db1c2947120d: Pull complete
00c1ea53d2dc: Pull complete
76b368b244f2: Pull complete
3a329ab4106c: Pull complete
135432843cbb: Pull complete
02f5fe1f230c: Pull complete
be615b4368c8: Pull complete
e1924c352a9e: Pull complete
6d53b28ea253: Pull complete
Digest: sha256:65a0c1467d5a132b9f0c3a1ffe664ea66172ab8602f23df8b5833bf4929b2192
Status: Downloaded newer image for couchbase/server:latest
95bff3b9294cb3d26100a916ad06b4bb50cfcc6a3476f7081d6bb9c47b4e8cd5
```

```
$ docker logs 95bff3b9294cb3d26100a916ad06b4bb50cfcc6a3476f7081d6bb9c47b4e8cd5
Starting Couchbase Server -- Web UI available at http://<ip>:8091
{"init terminating in do_boot",{{badmatch,{error,{bad_return,{{ns_babysitter,start,[normal,[]]},{'EXIT',{{worker_died,<0.71.0>,{badmatch,{error,eacces}}},{gen_server,call,['sink-babysitter_sink',{log,<<184 bytes>>},infinity]}}}}}}},[{init,start_it,1,[]},{init,start_em,1,[]}]}}
init terminating in do_boot ()
```

```
$ docker images                                                                                                                               
REPOSITORY                  TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
couchbase/server            latest              6d53b28ea253        5 days ago          803.8 MB
```

```
$ docker --version
Docker version 1.8.1, build d12ea79
```

**Comment by tleyden** [comment]
@bndw thanks for the issue report.

A few things to try:

Can you try it without the volume mount and see if that works?  (just want to level-set, we can come back to the volume mount)

**Comment by ceejatec** [comment]
That's a good point, Traun. The error mentions "eaccess" which would happen if the Couchbase user (UID 999 in the container) doesn't have write permission to /opt/couchbase/var.

I was able to produce a somewhat similar error on Linux by bind-mounting /opt/couchbase/var to an NFS-mounted directory, because on NFS, the local "root" user has no permissions. The docker daemon runs as root, so it was unable to chown the bind-mount directory to user 999. I'm guessing there may be a similar interaction with the Linux VM hosting docker on MacOS, although it is just a guess.

If this problem is as easy as that to trigger, we should figure out a suggested solution and update the image docs.

**Comment by bndw** [comment]
@tleyden It does work without the volume mount.

**Comment by bndw** [comment]
Any ideas? I've tried to chown the `~/couchbase` directory  for the couchbase user with `999:999`, but still have the same problem.

**Comment by hookenz** [comment]
By the way, it works perfectly under Ubuntu Linux with docker 1.8.1.  What distribution are you using?

Have you tried experimenting with the --user option?  Not sure if that may cause other issues but the ports it opens I think are high enough not to require root.
````
matthewh@slave:~$ docker run -d -v ~/couchbase:/opt/couchbase/var -p 8091:8091 couchbase/server
Unable to find image 'couchbase/server:latest' locally
latest: Pulling from couchbase/server
0cf7eebde823: Pull complete 
14a01c3766b5: Pull complete 
db1c2947120d: Pull complete 
00c1ea53d2dc: Pull complete 
76b368b244f2: Pull complete 
3a329ab4106c: Pull complete 
135432843cbb: Pull complete 
02f5fe1f230c: Pull complete 
be615b4368c8: Pull complete 
e1924c352a9e: Pull complete 
6d53b28ea253: Pull complete 
ba2b457ecfb2: Already exists 
26180b0fe8fb: Already exists 
edd7c1974a70: Already exists 
57bca5139a13: Already exists 
Digest: sha256:65a0c1467d5a132b9f0c3a1ffe664ea66172ab8602f23df8b5833bf4929b2192
Status: Downloaded newer image for couchbase/server:latest
2b9f71205e258c4c601b10894edc7d2394edd759f3d647f909e9607cc6a7e6a2
````
````
matthewh@slave:~$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                                                                           NAMES
2b9f71205e25        couchbase/server    "/entrypoint.sh couch"   11 seconds ago      Up 10 seconds       8092/tcp, 11207/tcp, 11210-11211/tcp, 0.0.0.0:8091->8091/tcp, 18091-18092/tcp   elated_wright
````
````
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                                                                           NAMES
2b9f71205e25        couchbase/server    "/entrypoint.sh couch"   2 minutes ago       Up 2 minutes        8092/tcp, 11207/tcp, 11210-11211/tcp, 0.0.0.0:8091->8091/tcp, 18091-18092/tcp   elated_wright
````
````
matthewh@slave:~$ docker logs elated_wright
Starting Couchbase Server -- Web UI available at http://<ip>:8091
````

**Comment by ceejatec** [comment]
Kind of grasping at straws here, but can you try using -v $HOME/couchbase:/opt/couchbase/var  rather than using ~ ?

Can you also tell us the boot2docker version you have? I don't know if that will auto-update along with Docker itself as I don't run on an OS that requires boot2docker. I do see that boot2docker versions prior to 1.3.0 would not automatically share /Users into the Virtualbox VM, which could account for this - but that version is almost a year old so I'd be surprised if you were still running it.

**Comment by ceejatec** [comment]
Another thing to try: go through the example using nginx on MacOS here: https://docs.docker.com/installation/mac/#mount-a-volume-on-the-container

That will help verify that normal mount-binding works in your Docker installation.

**Comment by arun-gupta** [comment]
+1 getting the same issue on Mac OSX with Docker Machine.

@ceejatec Tried the steps and they worked fine on my machine.

<img width="390" alt="screen shot 2015-10-14 at 7 29 52 pm" src="https://cloud.githubusercontent.com/assets/113947/10503359/029d28fc-72aa-11e5-9b8e-807138420ff9.png">
<img width="331" alt="screen shot 2015-10-14 at 7 29 57 pm" src="https://cloud.githubusercontent.com/assets/113947/10503360/029eceb4-72aa-11e5-93cc-f7c911f833ba.png">

**Comment by arun-gupta** [comment]
Hacked with @ceejatec 

Adding `RUN useradd couchbase -m -s /bin/bash` to Dockerfile got it working.

**Comment by ceejatec** [comment]
Yep, Arun and I sat together and figured it out. A little more detail:

VirtualBox is doing something kind of peculiar with the ownership of files in Shared Folders. Basically, outside the VM they appear as owned by the user running VirtualBox, while inside the boot2docker VM they appear as owned by the "docker" user, UID 1000. But they're locked to that UID, and cannot be changed - chown returns successfully but has no effect.

Unfortunately, the "couchbase" user inside the couchbase Docker container is UID 999. So it hits weird permissions issues and dies in various exciting ways.

The user 999 is actually being created as a side-effect of installing the Couchbase .deb package in the Dockerfile. However the .deb won't create a new "couchbase" user if one already exists. So the solution is to explicitly create that user with a UID of 1000 inside the Dockerfile. I'll be pushing a change to do that in just a minute.

This will fix the couchbase/server image. In order to fix the official Couchbase image, I'll also need to propose a change to the Docker guys, which I'll do in the next 24 hours. They usually approve those changes pretty quickly.

**Comment by ceejatec** [comment]
Fixed with 7d6d07ab7f0f20acdc1169e0154ce2fa41234f64.
